lib/saluki-io/src/net/util/retry/queue/persisted.rs

```diff
@@ -92,6 +134,14 @@ where
         self.entries.len()
     }
 
+    fn on_disk_bytes_limit(&self) -> Result<u64, GenericError> {
+        let total_space = self.disk_usage_retriever.inner.total_space()? as f64;
+        let available_space = self.disk_usage_retriever.inner.available_space()? as f64;
```

Comment:
Damn, I just realized after looking at `fs4` more closely that it only has natively async functions as an extension to types like `tokio::fs::File`. 😣

We'll want to change this function to be `async` and then wrap the entire body of the function with `tokio::task::spawn_blocking(async move { ... })`, and then await on the `JoinHandle<T>`.

This will also require us to make `DiskUsageRetrieverWrapper` cloneable (since we'll have to clone to send it to the blocking pool) and the contents it's wrapping.

Sorry about the initial confusion on `fs4` being natively async-friendly.

Reply:
Ah, sorry, yeah... I was being a doofus when I said `async move`.